app/controllers/chatfuel_controller.rb

```diff
@@ -1,12 +1,16 @@
 class ChatfuelController < ApplicationController
   def show
-    chatbot_channel = Channel.chatbot
+    @topic = Topic.find_by(name: params[:topic])
 
-    @text_component = Text::Sorter.sort(chatbot_channel.text_components, {}).first
+    if @topic
+      @text_component = Text::Sorter.sort(@topic.text_components, {}).first
```

Comment:
You are right, I did not take this into consideration, but only the text_components for the current topic which are assigned to the chatbot channel should be displayed.